### ISSUE TITLE
Save certificate to file

### ISSUE THREAD

**Issue by kelunik** [riebl/vanetza#13]
Do you have any guidance on how to save a `Certificate` instance to a file? I want to extend `socktap` to use a configured root certificate instead of a randomly generated one, so two instances can communicate with each other.

I have the rest of the code for generating one already, just saving + loading is missing. Any advice?

**Comment by riebl** [comment]
Hm, I haven't thought much about this before. Actually I see no big issue to use `serialize` and `deserialize` for writing/reading certificates from the file system as well. Basically any stream - including file streams - can be wrapped into `OutputArchive` or `InputArchive`, see common/serialization_buffer.hpp where in-memory archives are created.
Does this help you?

**Comment by kelunik** [comment]
I've looked into those, they seem to be the only serialization implemented right now. I couldn't find a standard format for C2X certificates yet.

In fact, my code from earlier today seems to be fully working, I just forgot a couple of `std::` prefixes and didn't realize that. Seems like a few hours of pause solved that. :smile: 

Closing for now, will report back if something doesn't work as expected.
